packages/vincent-contracts/src/facets/VincentAppFacet.sol

```diff
@@ -171,33 +173,41 @@ contract VincentAppFacet is VincentBase {
         VincentAppStorage.AppStorage storage as_ = VincentAppStorage.appStorage();
         VincentAppStorage.App storage app = as_.appIdToApp[appId];
 
-        // Check that no app versions have delegated agent PKPs
-        for (uint256 i = 0; i < app.appVersions.length; i++) {
-            if (app.appVersions[i].delegatedAgentPkps.length() > 0) {
-                revert LibVincentAppFacet.AppVersionHasDelegatedAgents(appId, i + 1);
-            }
-        }
-
         app.isDeleted = true;
         emit LibVincentAppFacet.AppDeleted(appId);
     }
 
+    /**
+     * @notice Undeletes an app by setting its isDeleted flag to false
+     * @dev Only the app manager can undelete an app
+     * @param appId ID of the app to undelete
+     */
+    function undeleteApp(uint256 appId) external onlyAppManager(appId) onlyRegisteredApp(appId) {
+        VincentAppStorage.AppStorage storage as_ = VincentAppStorage.appStorage();
+        VincentAppStorage.App storage app = as_.appIdToApp[appId];
+
+        app.isDeleted = false;
+        emit LibVincentAppFacet.AppUndeleted(appId);
+    }
+
```

Comment:
nit: require `app.isDeleted` so that `AppUndeleted` can't be emitted if the App has already been undeleted